### PR TITLE
PAYARA-3355 HK2 JDK-11 support

### PR DESCRIPTION
**Pull request by jGauravGupta**
Signed-off-by: Gaurav Gupta <gaurav.gupta.jc@gmail.com>

### CODE REVIEW

**Comment by Cousjava** [comment]
CQ 18517 http://dev.eclipse.org/ipzilla/show_bug.cgi?id=18517 for ASM 7.0

File: pom.xml [hunk 1]
```diff
@@ -139,7 +139,8 @@
         <jakarta.annotation.version>1.3.4</jakarta.annotation.version>
         <jakarta.json.version>1.1.5</jakarta.json.version>
         <glassfish.json.version>1.1.5</glassfish.json.version>
-        <javax.xml.bind.jaxb.version>2.1</javax.xml.bind.jaxb.version>
+        <jaxb-api.version>2.4.0-b180830.0359</jaxb-api.version>
+        <jaxb-runtime.version>2.4.0-b180830.0438</jaxb-runtime.version>
```
[the remark this answers is not in the document]
Comment (jGauravGupta) [reply pom.xml]: Updated to v2.3.1 as artifact javax.xml.bind:jaxb-api:jar:2.3.2 is not available in sonatype-nexus-staging.

File: hk2-core/pom.xml [hunk 2]
```diff
@@ -50,6 +50,13 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
+                <configuration>
+                    <argLine>-enableassertions --add-exports java.base/jdk.internal.loader=ALL-UNNAMED</argLine>
```
Comment (Cousjava) [inline hk2-core/pom.xml]:
```suggestion
                     <argLine>-XX:+IgnoreUnrecognizedVMOptions -enableassertions --add-exports="java.base/jdk.internal.loader=ALL-UNNAMED"</argLine>
```

Comment (Cousjava) [reply hk2-core/pom.xml]: This allows for JDK8

**Pull request merged**